enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,458 @@
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+	// The DeviceDiscoveryPolicies that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceDiscoveryPolicyType = "disk"
+	// Part represents a device-type of partion
+	Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+	// DeviceTypes that should be used for automatic detection. This would be one of the types supported
+	// by the local-storage operator. Currently the supported types are: disk,part
+	// If the list is empty the default value will be `[disk]`.
+	DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type AutoDetectVolumeStatus struct {
+	Status string `json:"status"`
+}
+```
+### Workflow of LocalPV creation via AutoDetectVolume CR
+- Once the `AutoDetectVolume` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
+- Possible `status.phase` values for `AutoDetectVolume`
+  - `Discovering`
+  - `Failed`
+  - `Discovered`
+  - `Provisioned`
+  
+Example of `AutoDetectVolume` CR:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: AutoDetectVolume
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+  storageClassName: example-storageclass
+  volumeMode: Block
+  minDeviceCount: 5
+  maxDeviceCount: 10
+  deviceInclusionSpec:
+    deviceTypes:
+      - disk
+    deviceMechanicalProperty: Rotational
+    minSize: 10G
+    maxSize: 100G
+Status:
+  Phase: Provisioned
```

Comment:
Since diskmaker daemon would periodically scan for new disks that match configured policy, what would "provision" phase mean? LSO could be in "provisioned" state on one node and "discovering" state on another node.

Reply:
Is diskmaker the one that will actually be creating the PVs? If so, seems like the only status we can show here is whether or not diskmaker has been notified on all the expected nodes.

Reply:
How about changing it to `waiting` or `notified`.

Reply:
resolving this since its been taken care in another PR #247